small-talk/app/components/ThemeLayout.js

```diff
@@ -1,12 +1,18 @@
-import React from "react";
+import React, { useState } from "react";
 import TopBar from "./TopBar";
 import SideBar from "./SideBar";
 
 const ThemeLayout = ({ children }) => {
+    const [isSidebarExpanded, setIsSidebarExpanded] = useState(false);
+
+    const toggleSidebar = () => {
+        setIsSidebarExpanded(!isSidebarExpanded);
+    };
+
     return(
         <div className="w-full min-h-screen flex flex-col">
-            <TopBar/>
-            <SideBar/>
+            <TopBar toggleSidebar={toggleSidebar}/>
+            <SideBar isExpanded={isSidebarExpanded}/>
```

Comment:
@39otsu Are we able to trigger this sidebarexpansion when working on keeping things responsive? I guess the big change would be to add another parameter that checks if the sidebar should be hidden in case of mobile.